synapse/replication/slave/storage/account_data.py

```diff
@@ -69,38 +69,25 @@ def stream_positions(self):
         result["tag_account_data"] = position
         return result
 
-    def process_replication(self, result):
-        stream = result.get("user_account_data")
-        if stream:
-            self._account_data_id_gen.advance(int(stream["position"]))
-            for row in stream["rows"]:
-                position, user_id, data_type = row[:3]
-                self.get_global_account_data_by_type_for_user.invalidate(
-                    (data_type, user_id,)
-                )
-                self.get_account_data_for_user.invalidate((user_id,))
+    def process_replication_rows(self, stream_name, token, rows):
+        if stream_name == "tag_account_data":
```

Comment:
we seem to have starrted with `user_account_data`, `room_account_data`, and `tag_account_data` (in that order), and have ended up with `tag_account_data` and `account_data`.

I *guess* `account_data` is a combination of `user_account_data` and `room_account_data`; but I'm also confused by the reordering.

This change might make sense, but it's the sort of thing I'd prefer to see happening independently of the change of protocol.